api/server/models/user.js

```diff
@@ -281,4 +281,36 @@ module.exports = function (User) {
     );
 //********************************* END FORGET PASSWORD **********************************
 
+//********************************* START RESET PASSWORD **********************************
+    User.reset_password = function ( req, password, callback) {
+        var access_token_userid = req.accessToken.userId;
+        User.findById( access_token_userid, function(err, useruser) {
+            if( err ){
+                callback(null, 0, 'Error', {});
+            }else{
+                useruser.updateAttribute('password', password, function(err, user) {
```

Comment:
we need user object to update the attribute.
if invalid acesstoken is there then api will itself throw unauthorised error, and if token is not found the it will throw error callback(null, 0, 'Error', {});